### ISSUE TITLE
`clean_unused_page_cache_pixmap` doesn't work as intended.

### ISSUE THREAD

**Issue by shenlebantongying** [emacs-eaf/eaf-pdf-viewer#76]
![image](https://user-images.githubusercontent.com/20123683/170467720-1f2e5b56-a0ad-4603-8204-20b899518f4d.png)

Open a lengthy PDF (hundreds of pages) and hold space bar ⇒ Memory usage will be increasing until a maximum value (all pages rendered and hold in memory). After rendering a few hundreds of pages, eaf-pdf-viewer will consume > 1 GiB memory.

I thought there is no handling of old pages that users cannot see, and it turns out to be one.

However, it doesn't seem to work?

It is called here:
https://github.com/emacs-eaf/eaf-pdf-viewer/blob/e534ffd128f0c1e13edd5fa88be41ee3ba50850c/eaf_pdf_widget.py#L407-L408

And the code only pop the oldest one.
https://github.com/emacs-eaf/eaf-pdf-viewer/blob/e534ffd128f0c1e13edd5fa88be41ee3ba50850c/eaf_pdf_widget.py#L323-L332

**Comment by manateelazycat** [comment]
PR are welcome.

**Comment by luhuaei** [comment]
How many memory usage if you only use fitz to load this large pdf file? @shenlebantongying 

```python
import os
import fitz

doc = fitz.open("mupdf_explored.pdf") # replace your pdf file.

with open("/proc/%d/status" % os.getpid(), "r") as f:
    for line in f.readlines():
        if line.startswith("VmSize"):
            print(line)
```

**Comment by shenlebantongying** [comment]
This file 
https://ca1lib.org/book/18159064/6c1d80

will yield 0.21 gB:

```
> python3 ./test.py
VmSize:   206352 kB
```

I don't think it is file-specific, I tried multiple files, the memory usage will only go up and never shrink. Hold the space bar and scrolling throw a few hundreds of pages will make this problem obvious.

I am using python3.10, master branch of eaf on archlinux.

**Comment by luhuaei** [comment]
`clean_unused_page_cache_pixmap` no clear PdfPage cache, I try to fix it on bbe49bc. @shenlebantongying

**Comment by shenlebantongying** [comment]
Marvelous! I tried the fix, and can confirm the memory footage now can keep a very low and constant value :D
